### PR TITLE
22846 - show resolution dates in amalgamations

### PR DESCRIPTION
**Pull request by ketaki-deodhar**
*Issue #:* /bcgov/entity#22846

*Description of changes:*

-  fetch resolutions
-  show resolution dates on summary page
-  save resolution dates in filing
- WIP - add more unit tests

Short form amalgamation (Horizontal):

image

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the bcrs-entities-create-ui license (Apache 2.0).

### CODE REVIEW

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://business-create-dev--pr-745-9wwaez00.web.app

**Review by ArwenQin — APPROVED** [review]
Looks good!

**Comment by severinbeauvais** [comment]
While you're working on this component, Jacqueline would like the Share Structure table headers to have the same format as the People and Roles table headers (#212559, bold, 14px). Can you make this change or should that be another ticket?

**Comment by ketaki-deodhar** [comment]
> While you're working on this component, Jacqueline would like the Share Structure table headers to have the same format as the People and Roles table headers (#212559, bold, 14px). Can you make this change or should that be another ticket?

No need for another ticket. I will do it here

File: src/components/common/ListResolutions.vue [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+      <v-row no-gutters>
+        <v-col
+          cols="12"
+          sm="3"
+          class="pr-4"
+        />
+        <v-col
+          cols="12"
+          sm="9"
+          class="pt-4 pt-sm-0"
```
Comment (severinbeauvais) [inline src/components/common/ListResolutions.vue:39]: On a narrow screen, pt-4 is too much.

image

I think you can use pt-0 or maybe it's not even needed.

File: src/interfaces/filing-interfaces/filing-interfaces.ts [hunk 2]
```diff
@@ -60,6 +61,8 @@ export interface AmalgamationFilingIF {
     incorporationAgreement?: {
       agreementType: string
     }
+    resolutions?: ResolutionIF[]
```
Comment (severinbeauvais) [inline src/interfaces/filing-interfaces/filing-interfaces.ts:64]: It is optional, or could you just save `[]` if there aren't any resolutions? Which do you think is better?

File: src/mixins/amalgamation-mixin.ts [hunk 3]
```diff
@@ -390,10 +391,12 @@ export default class AmalgamationMixin extends Vue {
     // NB - addresses and auth info have already been fetched (and checked above)
     // NB - make all API calls concurrently without rejection
     // NB - if any call failed, that item will be null
-    const [ directors, shareStructure ] =
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
```
Comment (severinbeauvais) [inline src/mixins/amalgamation-mixin.ts:394]: Do you still need this eslint override?

File: src/store/state/state-model.ts [hunk 4]
```diff
@@ -237,5 +237,6 @@ export const stateModel: StateModelIF = {
   },
   completingParty: null,
   parties: null,
+  resolutions: null,
```
Comment (severinbeauvais) [inline src/store/state/state-model.ts:240]: Or `[]`? Which is better?

File: src/mixins/amalgamation-mixin.ts (hunk 3 above)
Comment (ketaki-deodhar) [reply src/mixins/amalgamation-mixin.ts:394]: Nope. Forgot to remove. Will do it!

File: src/store/state/state-model.ts (hunk 4 above)
Comment (ketaki-deodhar) [reply src/store/state/state-model.ts:240]: I think `[]` is better. what would you suggest?

File: src/interfaces/filing-interfaces/filing-interfaces.ts (hunk 2 above)
Comment (ketaki-deodhar) [reply src/interfaces/filing-interfaces/filing-interfaces.ts:64]: save `[]` approach is better

Comment (severinbeauvais) [reply src/interfaces/filing-interfaces/filing-interfaces.ts:64]: I agree. Remove the optional character, initialize it as `[]` in the store state, and make sure saving/filing an empty list works.

File: src/store/state/state-model.ts (hunk 4 above)
Comment (severinbeauvais) [reply src/store/state/state-model.ts:240]: Agreed. See my response above. Resolving.

File: src/components/common/ListShareClass.vue [hunk 5]
```diff
@@ -39,7 +39,23 @@
         disable-pagination
         disable-sort
         hide-default-footer
+        hide-default-header
```
Comment (ketaki-deodhar) [inline src/components/common/ListShareClass.vue:42]: used `hide-default-header` to apply custom style.

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://business-create-dev--pr-745-9wwaez00.web.app

File: src/components/common/ListShareClass.vue [hunk 6]
```diff
@@ -39,7 +39,23 @@
...
+              <th
+                v-for="(h, index) in headers"
+                :key="index"
+                :class="h.class"
+              >
+                <span>{{ h.text }}</span>
+              </th>
+            </tr>
+          </thead>
+        </template>
```
Comment (ketaki-deodhar) [inline src/components/common/ListShareClass.vue:58]: As this is a common component, this is how it looks when just a list:
Before/DEV:
image

After:
image

When in edit mode:
Before/DEV:
image

After:
image

File: src/components/common/ListResolutions.vue [hunk 7]
```diff
@@ -13,13 +13,12 @@
...
           <label
             id="resolution-label"
-            class="font-weight-bold"
           >Resolution or court Order Dates</label>
         </v-col>
         <v-col
           cols="12"
           sm="9"
-          class="pt-4 pt-sm-0"
+          class="pt-sm-0"
```
Comment (severinbeauvais) [inline src/components/common/ListResolutions.vue:21]: Are you sure you need this?

Comment (ketaki-deodhar) [reply src/components/common/ListResolutions.vue:21]: Testing.....

**Comment by ketaki-deodhar** [comment]
> > image
> 
> I think the font size should be 14px, same as the other content in the table.
> 
> Also, the label colour should be #212529 and the value colour should be #495057.
> 
> You can confirm with Jacqueline if you want.

Also, Jacqueline to take a look at this

File: src/components/common/ListShareClass.vue [hunk 8]
```diff
@@ -50,10 +66,18 @@
...
+            </td>
+            <td class="share-series-value">
+              {{ row.item.parValue ? row.item.parValue : 'No Par Value' }}
+            </td>
+            <td class="share-series-value">
+              {{ row.item.currency }}
+            </td>
+            <td class="share-series-value">
+              {{ row.item.hasRightsOrRestrictions ? 'Yes' : 'No' }}
+            </td>
```
Comment (ketaki-deodhar) [inline src/components/common/ListShareClass.vue:80]: Jacqueline took a look at this and suggested to change the color for values

**Comment by jacqueline-williams-549** [comment]
> > > image
> > 
> > 
> > I think the font size should be 14px, same as the other content in the table.
> > Also, the label colour should be #212529 and the value colour should be #495057.
> > You can confirm with Jacqueline if you want.
> 
> Also, Jacqueline to take a look at this

Reviewed with Ketaki and getting back to her about suggestion about language

File: src/components/common/ListShareClass.vue (hunk 6 above)
Comment (severinbeauvais) [reply src/components/common/ListShareClass.vue:58]: Awesome! Looks great.

**Review by severinbeauvais — APPROVED** [review]
LGTM. Remember to update the app version.

**Pull request merged**